### PR TITLE
Add jest mock for the Linking library

### PR DESCRIPTION
**Pull request by Maxwell2022**
## Motivation

This PR is adding a mock of the React Native library `Linking`. Some of the third party library are expecting this Module to be mocked. If not mocked, jest test is throwing the following error:

>   ● Test suite failed to run
>
>     Invariant Violation: Native module cannot be null.
> 
>       at invariant (node_modules/fbjs/lib/invariant.js:44:15)
>       at Linking.NativeEventEmitter (node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter.js:32:1)
>       at new Linking (node_modules/react-native/Libraries/Linking/Linking.js:119:141)
>       at Object.<anonymous> (node_modules/react-native/Libraries/Linking/Linking.js:191:16)
>       at Object.get Linking [as Linking] (node_modules/react-native/Libraries/react-native/react-native.js:90:22)
>       at Object.<anonymous> (node_modules/react-native-experimental-navigation/NavigationRootContainer.js:15:36)

## Test Plan

There are no tests for mocks. Actual tests are not broken after adding this mock

## Additional Information

The current workaround is to have your own `jest-setup.js`, that need to be added in the `jest` configuration, with the following mock:

```
jest.mock('Linking', () => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  openURL: jest.fn(),
  canOpenURL: jest.fn(),
  getInitialURL: jest.fn(),
}));
```

And in your `package.json`:

```
"jest": {
  "preset": "react-native",
  "setupFiles": [
    "./node_modules/react-native/jest/setup.js",
    "./jest-setup.js"
  ]
}
```

### CODE REVIEW

File: jest/setup.js [hunk 1]
```diff
@@ -129,6 +129,13 @@ const mockNativeModules = {
     addListener: jest.fn(),
     removeListeners: jest.fn(),
   },
+  Linking: {
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
```
Comment (javache) [inline jest/setup.js]: Please mock `startObserving` and `stopObserving` instead of these two.

**Comment by javache** [comment]
Will need to be rebased on https://github.com/facebook/react-native/pull/12839

File: jest/setup.js (hunk 1 above)
Comment (Maxwell2022) [inline jest/setup.js]: I'm confused, these methods are not exposed in the `Linking` library:
`Libraries/Linking/Linking.js:127`

**Comment by javache** [comment]
Looks like something went wrong with your rebase. Please verify your diff looks correct.

File: jest/setup.js (hunk 1 above)
Comment (javache) [inline jest/setup.js]: No, but they are on the NativeModule (and used by EventEmitter)

**Comment by Maxwell2022** [comment]
That's weird, when I rebase my branch to master it says that I'm 84 ahead, 69 behind.
I just ran `git rebase master jest-mock-linking`

Not sure whatI'm missing here

**Comment by Maxwell2022** [comment]
I don't know why it's messed up but I think this PR doesn't make sense anymore as it has been implemented in other PRs. Just closing it

**Pull request closed without merging**